### PR TITLE
feat: add Config.Validate() to allow validation of configuration options before starting the manager

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

This PR adds `Config.Validate()` to allow validation of configuration options before starting the manager.

This will also set the stage to allow stopping the manager boot up process when mutually exclusive options are being provided (e.g. as in single controller deployments `kong-admin-url` and potential future Admin API service name specified for service discovery).

Additionally this performs some minor refactor around config and passing down the configuration options to downstream consumers of said options. It also stores the flagset in the `Config` so that we can use `c.flagSet.Changed()` to verify if the flag has been provided by the user ( in case the default flag value is a non empty/default value).

### CODE REVIEW

File: internal/manager/config.go [hunk 1]
```diff
@@ -101,12 +104,37 @@ type Config struct {
...
+	flagSet *pflag.FlagSet
 }
 
 // -----------------------------------------------------------------------------
 // Controller Manager - Config - Methods
 // -----------------------------------------------------------------------------
 
+// Validate validates the config. With time this logic may grow to invalidate
+// incorrect configurations.
+func (c *Config) Validate() error {
```
Comment (czeslavo) [inline internal/manager/config.go:117]: Would you mind adding a unit test that would verify the validation rules work as expected? There could be an example valid Config, and test cases could modify its fields to trigger the error.

File: internal/manager/config.go [hunk 2]
```diff
@@ -66,9 +68,10 @@ type Config struct {
 	GatewayAPIControllerName string
 
 	// Ingress status
-	PublishService       string
-	PublishStatusAddress []string
-	UpdateStatus         bool
+	PublishService              string
+	publishServiceNamespaceName types.NamespacedName
```
Comment (czeslavo) [inline internal/manager/config.go:72]: Adding a private field that duplicates the already existing fields seems fishy to me. I'm wondering if we wouldn't like to split this by e.g. leaving the `Config` struct to carry the raw flags that are set by the user (as it is now, no validation whatsoever), and creating a second one (could be named `ValidatedConfig`) that would have a constructor accepting the old `Config`, doing the validations in it. That gives us a clear way to tell that if we accept the `ValidatedConfig` somewhere, we can be sure the data is valid. WDYT?

Comment (czeslavo) [reply internal/manager/config.go:72]: If you find it too much work to tackle it in this PR, I think it still could be worth creating an issue for that - probably there are more cases in which we use some fields from the `Config` and we have to validate them on the go while we could have them checked beforehand.

**Review by czeslavo — COMMENTED** [review]
I like the direction of making the validation run as soon as possible, not leaving the responsibility to the routines that accept the `Config`. It's gonna be less error-prone and will save us from scattering validation code over places that should not be responsible for that.

File: internal/manager/config.go (hunk 2 above)
Comment (pmalek) [reply internal/manager/config.go]: Thanks for the suggestion: added `FlagNamespacedName` to parse and validate command line flags straight into the config field. PTAL

Comment (czeslavo) [reply internal/manager/config.go]: Clever! 👍

**Review by czeslavo — APPROVED** [review]

**Pull request merged**